python-frontend/src/test/java/org/sonar/python/tree/PythonTreeMakerTest.java

```diff
@@ -747,6 +749,54 @@ void nonlocalStatement() {
     assertThat(nonlocalStatement.children()).hasSize(3);
   }
 
+  @Test
+  void funcdef_statement_type_params() {
+    setRootRule(PythonGrammar.FUNCDEF);
+    var astNode = p.parse("def overly_generic[\n" +
+      "   SimpleTypeVar,\n" +
+      "   TypeVarWithBound: int,\n" +
+      "   TypeVarWithConstraints: (str, bytes),\n" +
+      "   *SimpleTypeVarTuple,\n" +
+      "   **SimpleParamSpec\n" +
+      "](\n" +
+      "   a: SimpleTypeVar,\n" +
+      "   b: TypeVarWithBound,\n" +
+      "   c: Callable[SimpleParamSpec, TypeVarWithConstraints],\n" +
+      "   *d: SimpleTypeVarTuple,\n" +
+      "): pass");
+    var functionDef = treeMaker.funcDefStatement(astNode);
```

Comment:
I suggest to refactor this way:

```suggestion
    var functionDef = parse("def overly_generic[\n" +
      "   SimpleTypeVar,\n" +
      "   TypeVarWithBound: int,\n" +
      "   TypeVarWithConstraints: (str, bytes),\n" +
      "   *SimpleTypeVarTuple,\n" +
      "   **SimpleParamSpec\n" +
      "](\n" +
      "   a: SimpleTypeVar,\n" +
      "   b: TypeVarWithBound,\n" +
      "   c: Callable[SimpleParamSpec, TypeVarWithConstraints],\n" +
      "   *d: SimpleTypeVarTuple,\n" +
      "): pass", treeMaker::funcDefStatement);
```

When calling `RuleTest#parse`, a few additional things are done to ensure every child is correctly visited and that they are in the correct order. I think it makes sense to have here because currently, removing the sorting of the children of `TypeParams` doesn't crash anything (but it does when using this test method).